c/array-patterns/array14_pattern.c

```diff
@@ -49,8 +49,8 @@ int main()
 	while(1)
         {
 		
-		index1 = __VERIFIER_nondet_short() ;
-		index2 = __VERIFIER_nondet_short() ;
+		index1 = (signed long long)__VERIFIER_nondet_short() ;
+		index2 = (signed long long)__VERIFIER_nondet_short() ;
```

Comment:
Which overflow are you trying to guard against - the `index1+index2`? I believe this will be subject to promotion to `int` anyway, and thus shouldn't overflow.

Reply:
As you mentioned, I wanted to guard it against index1+index2 overflow. I promoted the types of these indices just to be extra careful.